test/type/runtime_discoverable_attrs.swift

```diff
@@ -264,3 +277,44 @@ extension EnumFlag where B == Void {
   @EnumFlag func testInst() {}
   @EnumFlag static func testStatic() -> Int { 42 }
 }
+
+@Flag extension EnumTypeTest { // expected-error {{@Flag can only be applied to unavailable extensions in the same module as 'EnumTypeTest'}}
+}
+
+@available(*, unavailable)
+@Flag extension EnumTypeTest { // Ok
+}
+
+@available(*, unavailable)
+@EnumFlag extension EnumTypeTest { // expected-error {{@EnumFlag is already applied to type 'EnumTypeTest'; did you want to remove it?}} {{275:1-11=}}
```

Comment:
There is probably a reason for it, but why the fix-it is removing the original instead of this occurrence of `@EnumFlag`?

Reply:
Because if you are trying to add unavailable extension then you don't really want this flag on the declaration.

Reply:
In other words - these declarations are mutually exclusive and based on the extension it's always better to remove attribute from the type, one might also consider keeping the one on extension in place though if type is a class to opt-out the type and all it's sub-classes from use of the attribute. (I still need to implement a diagnostic for sub-classes).

Reply:
Ah I see, thanks for detailed explanation @xedin! I was more curious, because to me if one was trying to add an extension with attribute that already is on original decl, it doesn't need to be added on extension,so it would sounded reasonable remove from there, but I missed the availability bit.